app/controllers/page_views_controller.rb

```diff
@@ -57,4 +60,9 @@ def update_organic_page_views
       where("created_at > ?", 1.month.ago).sum(:counts_for_number_of_views)
     @article.update_column(:organic_page_views_past_month_count, organic_count_past_month_count) if organic_count_past_month_count > @article.organic_page_views_past_month_count
   end
+
+  def session_current_user_id
+    # custom defined here, because controller does not inherent from application_controller
+    session["warden.user.user.key"].flatten[0] if session["warden.user.user.key"].present?
```

Comment:
Rather than redefining it here can we just call it from the `ApplicationController`? That way we don't have to worry about these two methods getting out of sync.

Reply:
Would calling `ApplicationController` at all have the affect of loading in a bunch of allocations we're trying to avoid with this change?

I'm not sure I'm correct in thinking that is how it works.

I definitely agree that we want to DRY-ify this, but there's probably a more efficient third place we could put this method for all to use?

Reply:
ApplicationController is always going to be loaded bc its in the Rails default load paths. I think what you are hoping to avoid is actually calling any "default" methods that ApplicationController would put into the request path. As far as I know, those will not suddenly be included bc we call a single method on it.

The other option is the extract that simple method into a module in the concerns folder under controllers and call it from there.